pkg/validate/validate.go

```diff
@@ -2,39 +2,164 @@ package validate
 
 import (
 	"context"
+	"fmt"
 	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha1"
 	"github.com/armory/spinnaker-operator/pkg/halconfig"
+	"github.com/go-logr/logr"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
+	"sort"
 )
 
-// Transformers tracks the list of transformers
-var Validators []SpinnakerValidator
+// generators is used to register all ValidatorGenerator
+var generators []ValidatorGenerator
 
 func init() {
-	Validators = append(Validators, &singleNamespaceValidator{})
+	generators = append(generators, &singleNamespaceValidatorGenerator{}, &kubernetesAccountValidatorGenerator{})
 }
 
 type SpinnakerValidator interface {
-	Validate(svc v1alpha1.SpinnakerServiceInterface, hc *halconfig.SpinnakerConfig, options Options) error
+	GetName() string
+	GetPriority() Priority
+	Validate() ValidationResult
+}
+
+// Priority indicates the order in which the validator should be run, or if it doesn't matter (NoPreference).
+// validators with NoPreference = true can be run in parallel
+type Priority struct {
+	NoPreference bool
+	Order        int
+}
+
+type ValidationResult struct {
+	Error error
+	Fatal bool
+}
+
+type ValidatorGenerator interface {
+	Generate(svc v1alpha1.SpinnakerServiceInterface, hc *halconfig.SpinnakerConfig, options Options) ([]SpinnakerValidator, error)
 }
 
 type Options struct {
 	Ctx    context.Context
 	Client client.Client
 	Req    types.Request
+	Log    logr.Logger
+}
+
+type Account interface {
```

Comment:
This doesn't seem to be used anymore. The intent of that struct was to provide common utils for account based validation. Account A is validated -> store result with checksum in the status to avoid over-validating.